components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/JITProvisioningPostAuthenticationHandler.java

```diff
@@ -354,6 +366,44 @@ private boolean isUserNameFoundFromUserIDClaimURI(Map<String, String> localClaim
                 (localClaimValues.get(userIdClaimUriInLocalDialect));
     }
 
+    private boolean isAccountLocked(String username, String tenantDomain)
+            throws PostAuthenticationFailedException {
+
+        AccountLockService accountLockService = FrameworkServiceDataHolder.getInstance().getAccountLockService();
+        try {
+            return accountLockService.isAccountLocked(username, tenantDomain);
+        } catch (AccountLockServiceException e) {
+            throw new PostAuthenticationFailedException(
+                    ErrorMessages.ERROR_WHILE_CHECKING_ACCOUNT_LOCK_STATUS.getCode(),
+                    String.format(ErrorMessages.ERROR_WHILE_CHECKING_ACCOUNT_LOCK_STATUS.getMessage(), username), e);
+        }
+    }
+
+    private void handleAccountLockLoginFailure(String retryPage, AuthenticationContext context,
+                                               HttpServletResponse response) throws PostAuthenticationFailedException {
+
+        try {
+            // ToDo: Add support to configure enable/disable authentication failure reason.
+            boolean showAuthFailureReason = true;
+            StringBuilder urlBuilder = new StringBuilder();
+            urlBuilder.append(retryPage);
+            urlBuilder.append("?sp=").append(context.getServiceProviderName());
```

Comment:
better to use FrameworkUtils.appendQueryParams for this

Reply:
Fixed https://github.com/wso2/carbon-identity-framework/pull/3598/commits/44b14a3dc4c0be84122db9c92c09d08ce2daf877